src/economy/ferry_fleet.cc

```diff
@@ -237,7 +244,16 @@ void FerryFleet::remove_ferry(EditorGameBase& egbase, Ferry* ferry) {
 		              serial(), ferry ? ferry->serial() : 0);
 		return;
 	}
-	ferries_.erase(it);
+	while (it != ferries_.end()) {
+		ferries_.erase(it);
+		it = std::find(ferries_.begin(), ferries_.end(), ferry);
+		if (it != ferries_.end()) {
+			log_err_time(egbase.get_gametime(),
+			             "FerryFleet %u: Multiple instances of ferry %u were in the ferry fleet\n",
```

Comment:
Does this actually happen?
```suggestion
			log_warn_time(egbase.get_gametime(),
			             "FerryFleet %u: Multiple instances of ferry %u were in the ferry fleet",
```

Reply:
It shouldn't. But ships at least had this problem at some point, so I'm a little wary.